File: BotsSDK/korebot/src/main/java/kore/botssdk/adapter/CardTemplateListAdapter.java
package kore.botssdk.adapter;

import static android.view.View.GONE;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.util.Base64;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.squareup.picasso.Picasso;

import java.util.ArrayList;

import kore.botssdk.R;
import kore.botssdk.models.AdvanceListTableModel;
import kore.botssdk.utils.StringUtils;
import kore.botssdk.viewUtils.RoundedCornersTransform;

public class CardTemplateListAdapter extends RecyclerView.Adapter<CardTemplateListAdapter.ButtonViewHolder> {
    private final LayoutInflater inflater;
    private final ArrayList<AdvanceListTableModel.AdvanceTableRowDataModel> buttons;

    public CardTemplateListAdapter(@NonNull Context context, @NonNull ArrayList<AdvanceListTableModel.AdvanceTableRowDataModel> buttons) {
        this.buttons = buttons;
        this.inflater = LayoutInflater.from(context);
    }

    @NonNull
    @Override
    public CardTemplateListAdapter.ButtonViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        return new CardTemplateListAdapter.ButtonViewHolder(inflater.inflate(R.layout.card_desc_list_cell, viewGroup, false));
    }

    @Override
    public void onBindViewHolder(@NonNull CardTemplateListAdapter.ButtonViewHolder holder, int i) {
        AdvanceListTableModel.AdvanceTableRowDataModel btn = buttons.get(i);
        holder.tvBtnText.setText(btn.getTitle());

        holder.tvDescriptionTitle.setVisibility(GONE);
        holder.ivListBtnIcon.setVisibility(GONE);

        if(!StringUtils.isNullOrEmpty(btn.getDescription())) {
            holder.tvDescriptionTitle.setVisibility(View.VISIBLE);
            holder.tvDescriptionTitle.setText(btn.getTitle());
            holder.tvBtnText.setText(btn.getDescription());
        }

        if(!StringUtils.isNullOrEmpty(btn.getIcon()))
        {
            holder.ivListBtnIcon.setVisibility(View.VISIBLE);
            try {
                String imageData;
                imageData = btn.getIcon();
                if (imageData.contains(","))
                {
                    imageData = imageData.substring(imageData.indexOf(",") + 1);
                    byte[] decodedString = Base64.decode(imageData.getBytes(), Base64.DEFAULT);
                    Bitmap decodedByte = BitmapFactory.decodeByteArray(decodedString, 0, decodedString.length);
                    holder.ivListBtnIcon.setImageBitmap(decodedByte);
                }
                else
                {
                    Picasso.get().load(btn.getIcon()).transform(new RoundedCornersTransform()).into(holder.ivListBtnIcon);
                }
            }
            catch (Exception ex)
            {
                holder.ivListBtnIcon.setVisibility(GONE);
            }
        }
    }

    @Override
    public int getItemCount() {
        return buttons != null ? buttons.size() : 0;
    }

    boolean isFullView;

    public void setIsFromFullView(boolean isFullView) {
        this.isFullView = isFullView;
    }

    public static class ButtonViewHolder extends RecyclerView.ViewHolder {
        final TextView tvBtnText;
        final TextView tvDescriptionTitle;
        final ImageView ivListBtnIcon;

        public ButtonViewHolder(@NonNull View itemView) {
            super(itemView);
            tvBtnText = itemView.findViewById(R.id.tvDescription);
            tvDescriptionTitle = itemView.findViewById(R.id.tvDescriptionTitle);
            ivListBtnIcon = itemView.findViewById(R.id.ivListBtnIcon);
        }
    }
}

File: BotSdk/ui/src/main/java/com/kore/ui/adapters/CardTemplateListAdapter.kt
package com.kore.ui.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.kore.ui.utils.BitmapUtils
import com.kore.model.constants.BotResponseConstants
import com.kore.ui.R

class CardTemplateListAdapter(
    val context: Context,
    private val listItems: List<Map<String, *>>,
) : RecyclerView.Adapter<CardListViewHolder>() {
    override fun getItemCount(): Int {
        return listItems.size
    }

    override fun onBindViewHolder(holder: CardListViewHolder, position: Int) {
        val item = listItems[position]
        holder.tvBtnText.text = item[BotResponseConstants.KEY_TITLE] as String

        if (item[BotResponseConstants.DESCRIPTION] != null) {
            holder.tvDescriptionTitle.visibility = View.VISIBLE
            holder.tvDescriptionTitle.text = item[BotResponseConstants.KEY_TITLE] as String
            holder.tvBtnText.text = item[BotResponseConstants.DESCRIPTION] as String
        }

        if (item[BotResponseConstants.ICON] != null) {
            holder.ivListBtnIcon.visibility = View.VISIBLE
            try {
                val imageData = item[BotResponseConstants.ICON] as String
                BitmapUtils.loadImage(context, imageData, holder.ivListBtnIcon, R.drawable.ic_image_photo)
            } catch (ex: Exception) {
                holder.ivListBtnIcon.visibility = View.GONE
            }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CardListViewHolder {
        return CardListViewHolder(
            LayoutInflater.from(context).inflate(R.layout.card_desc_list_cell, parent, false)
        )
    }
}


class CardListViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    val tvBtnText: TextView
    val tvDescriptionTitle: TextView
    val ivListBtnIcon: ImageView

    init {
        tvBtnText = itemView.findViewById<TextView>(R.id.tvDescription)
        tvDescriptionTitle = itemView.findViewById<TextView>(R.id.tvDescriptionTitle)
        ivListBtnIcon = itemView.findViewById<ImageView>(R.id.ivListBtnIcon)
    }
}

